crypto/cms/cms_smime.c

```diff
@@ -622,6 +622,7 @@ CMS_ContentInfo *CMS_sign_receipt(CMS_SignerInfo *si,
     if (r)
         return cms;
     CMS_ContentInfo_free(cms);
+    ASN1_OCTET_STRING_free(os);
```

Comment:
Nort sure I understand the free here? 
*pos = os; means that it poitns to the data that you just freed??

Reply:
no, if that line is reached we also set r = 1, and therefore we dont reach
this line due to the if (r) return cms; above.

Reply:
I would probably have rewritten that to...

````
    BIO_free(rct_cont);
    return cms;
err:
    BIO_free(rct_cont);
    CMS_ContentInfo_free(cms);
    ASN1_OCTET_STRING_free(os);
    return NULL;
````